internal/stateindex/encoding_test.go

```diff
@@ -7,6 +7,40 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestEncodingInt64(t *testing.T) {
+	tests := []struct {
+		name string
+		n    int64
+	}{
+		{
+			name: "maximum negative value",
+			n:    -math.MaxInt64,
+		},
+		{
+			name: "minimum negative value",
+			n:    -1,
+		},
+		{
+			name: "maximum positive value",
+			n:    math.MaxInt64,
+		},
+		{
+			name: "minimum positive value",
+			n:    math.MaxInt64,
```

Comment:
min is 0, and add +1 as well

Reply:
I would also fill an array with a few more values just for good measure, like maybe +/- 3^k, k=1:39,
sort the array, encode, sort, decode, and compare